### PR TITLE
Test refactorings

### PR DESCRIPTION
**Pull request by timbussmann**
A few small test changes based on some randomly failing tests

### CODE REVIEW

File: src/NServiceBus.Core.Tests/Transports/Learning/AsyncFileTests.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public class AsyncFileTests
         public async Task When_file_content_larger_than_buffer_size()
         {
             var originalContent = string.Join("", Enumerable.Repeat("a#~×ψؾࢯ‽%1", 2000));
-            const string filePath = "test.txt";
```
Comment (timbussmann) [inline src/NServiceBus.Core.Tests/Transports/Learning/AsyncFileTests.cs]: this seems to be an issue when running via vstest. If we agree on this, we probably have to use the proposed solution on quite a few more tests

File: src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs [hunk 2]
```diff
@@ -255,7 +255,7 @@ ILearningTransportTransaction GetTransaction()
                 {
                     await transaction.Commit()
                         .ConfigureAwait(false);
-
+                    log.Debug($"Dropping message '{messageId}' as the specified TimeToBeReceived of '{ttbrString}' expired since sending the message at '{sentTime:O}'");
```
Comment (timbussmann) [inline src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs]: I think this should actually be Info instead of Debug. Thoughts?

Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs]: 👍 for Info

File: src/NServiceBus.Core.Tests/Transports/Learning/AsyncFileTests.cs (hunk 1 above)
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Transports/Learning/AsyncFileTests.cs]: You're saying that TestContext.CurrentContext.TestDirectory doesn't work for vstest?

Comment (timbussmann) [inline src/NServiceBus.Core.Tests/Transports/Learning/AsyncFileTests.cs]: sorry, I meant that the old approach is not working for vstests. But we have quite a few more tests which fail due to that reason, so that's a change we should extract into a dedicated PR anyways

**Comment by timbussmann** [comment]
@andreasohlund should be ready now
